test/input_gen/genInput.py

```diff
@@ -351,16 +411,55 @@ def bn_tf(x, *, trainable=True, init_beta=gen_tensor, init_gamma=gen_tensor, axi
+    x=gen_input(base_name+"_conv2DLayer.in", [i_b, i_c, i_h, i_w], int_data=True)
+    kernel=gen_input(base_name+"_conv2DKernel.in", [k_c, i_c, k_h, k_w], int_data=True)
     with open(base_name+"_conv2DKernel.in", 'ab') as outfile:
         np.array(bias, dtype=np.float32).tofile(outfile)
 
-    golden_conv, golden_grad_input, golden_grad_kernel, golden_grad_bias=conv2d_tf(x, kernel, i_b, i_h, i_w, i_c, k_h, k_w, k_c, stride, padding, bias)
+    golden_conv, golden_grad_input, golden_grad_kernel, golden_grad_bias, \
+        golden_conv2, golden_grad_input2, golden_grad_kernel2, golden_grad_bias2 = \
+        conv2d_tf(x, kernel, i_b, i_h, i_w, i_c, k_h, k_w, k_c, stride, padding, bias)
     save(base_name+"_goldenConv2DResult.out", np.transpose(golden_conv,(0,3,1,2)))
     save(base_name+"_goldenInputGrad.out", np.transpose(golden_grad_input,(0,3,1,2)))
     save(base_name+"_goldenKernelGrad.out", np.transpose(golden_grad_kernel,(3,2,0,1)))
     save(base_name+"_goldenBiasGrad.out", golden_grad_bias)
+    save(base_name+"_goldenConv2DResult2.out", np.transpose(golden_conv2,(0,3,1,2)))
+    save(base_name+"_goldenInputGrad2.out", np.transpose(golden_grad_input2,(0,3,1,2)))
+    save(base_name+"_goldenKernelGrad2.out", np.transpose(golden_grad_kernel2,(3,2,0,1)))
+    save(base_name+"_goldenBiasGrad2.out", golden_grad_bias2)
+
+def gen_test_case_conv_int(base_name):
+    x=gen_input(base_name+"_conv2DLayer.in", [1, 3, 28, 28], int_data=True)
+    kernel=gen_input(base_name+"_conv2DKernel.in", [6, 3, 5, 5], int_data=True)
+    bias = np.ones(6)
+    with open(base_name+"_conv2DKernel.in", 'ab') as outfile:
+        np.array(bias, dtype=np.float32).tofile(outfile)
+    kernel2=gen_input(base_name+"_conv2DKernel2.in", [12, 6, 1, 1], int_data=True)
+    bias2 = np.ones(12)
+    with open(base_name+"_conv2DKernel2.in", 'ab') as outfile:
+        np.array(bias2, dtype=np.float32).tofile(outfile)
+
+    golden_conv, golden_conv2, golden_grads = conv2d_tf_2(x, kernel, bias, kernel2, bias2)
+    save(base_name+"_goldenConv2DResult.out", np.transpose(golden_conv,(0,3,1,2)))
+    save(base_name+"_goldenConv2DResult2.out", np.transpose(golden_conv2,(0,3,1,2)))
+    save(base_name+"_goldenInputGrad.out", np.transpose(golden_grads[0],(0,3,1,2)))
+    save(base_name+"_goldenKernelGrad.out", np.transpose(golden_grads[1],(3,2,0,1)))
+    save(base_name+"_goldenBiasGrad.out", golden_grads[2])
+    save(base_name+"_goldenKernel2Grad.out", np.transpose(golden_grads[3],(3,2,0,1)))
+    save(base_name+"_goldenBias2Grad.out", golden_grads[4])
+
+def gen_test_case_conv_3(base_name):
+    x=gen_input(base_name+"_conv2DLayer.in", [1, 6, 24, 24], int_data=True)
+    kernel=gen_input(base_name+"_conv2DKernel.in", [12, 6, 5, 5], int_data=True)
+    bias = np.ones(12)
+    with open(base_name+"_conv2DKernel.in", 'ab') as outfile:
+        np.array(bias, dtype=np.float32).tofile(outfile)
+
+    golden_conv, golden_grads = conv2d_tf_2(x, kernel, bias, kernel, bias)
```

Comment:
I think this unittest can just be removed
it doesn't do anything new

Reply:
right.